src/Acme.UI/Models/Authentication/MemberViewModel.cs

```diff
@@ -0,0 +1,201 @@
+                return RegisterModel.Name;
+            }
+            set
+            {
+                RegisterModel.Name = value;
+            }
+        }
+
+        public string UserName
+        {
+            get
+            {
+                return RegisterModel.Username;
+            }
+            set
+            {
+                RegisterModel.Username = value;
+            }
+        }
+
+        public string Email
+        {
+            get
+            {
+                return RegisterModel.Email;
+            }
+            set
+            {
+                RegisterModel.Email = value;
+            }
+        }
+
+        public string Password
+        {
+            get
+            {
+                return RegisterModel.Password;
+            }
+            set
+            {
+                RegisterModel.Password = value;
+            }
+        }
+
+        public string Address
+        {
+            get
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == addressAlias);
+                if (found != null)
+                {
+                    return found.Value;
+                }
+                return null; 
+            }
+
+            set
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == addressAlias);
+                if (found != null)
+                {
+                    RegisterModel.MemberProperties.Remove(found);
+                }
+                RegisterModel.MemberProperties.Add(new UmbracoProperty { Alias = addressAlias, Value = value});
+            }
+        }
+
+        public string Phone
+        {
+            get
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == phoneAlias);
+                if (found != null)
+                {
+                    return found.Value;
+                }
+                return null;
+            }
+
+            set
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == phoneAlias);
+                if (found != null)
+                {
+                    RegisterModel.MemberProperties.Remove(found);
+                }
+                RegisterModel.MemberProperties.Add(new UmbracoProperty { Alias = phoneAlias, Value = value });
+            }
+        }
+
+        public bool IsApproved
+        {
+            get
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == umbracoIsApprovedPropAlias);
+                if (found != null)
+                {
+                    return found.Value != "0";
+                }
+                return true; // it is true in all other cases while registering the user
+            }
+
+            set
+            {
+                var found = RegisterModel.MemberProperties.FirstOrDefault(e => e.Alias == umbracoIsApprovedPropAlias);
```

Comment:
I dont think you need to Linq Query the properties, I suspect all of these are associative arrays that you can index by 
MemberProperties["myPropertyName"]

Please go look for all instances you are doing .FirstOrDefault and update them if you can make it simpler